cmd/cluster/instance_migrate_to_default_ingress.go

```diff
@@ -0,0 +1,555 @@
+	Long:    "Migrate the AstarteVoyagerIngress resource to an equivalent AstarteDefaultIngress",
+	Example: `  astartectl cluster migrate-ingress --namespace astarte --operator-name astarte-operator-controller-manager --operator-namespace kube-system -o avi.bkp`,
+	RunE:    migrateIngressF,
+}
+
+func init() {
+	migrateIngressCmd.PersistentFlags().String("operator-name", "astarte-operator-controller-manager", "The name of the Astarte Operator instance.")
+	migrateIngressCmd.PersistentFlags().String("operator-namespace", "kube-system", "The namespace in which the Astarte Operator resides.")
+	migrateIngressCmd.PersistentFlags().StringP("out", "o", "", "The name of the file in which the AVI custom resource will be saved.")
+
+	InstancesCmd.AddCommand(migrateIngressCmd)
+}
+
+func migrateIngressF(command *cobra.Command, args []string) error {
+	astarteNamespace, err := command.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+	operatorName, err := command.Flags().GetString("operator-name")
+	if err != nil {
+		return err
+	}
+	operatorNamespace, err := command.Flags().GetString("operator-namespace")
+	if err != nil {
+		return err
+	}
+
+	// is the migration allowed?
+	if err := ensureOperatorMinimumVersionRequirement(operatorName, operatorNamespace); err != nil {
+		return err
+	}
+
+	// ensure the Voyager CRD is present
+	_, err = kubernetesAPIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(),
+		"ingresses.voyager.appscode.com", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	aviList, err := kubernetesDynamicClient.Resource(aviV1Alpha1).Namespace(astarteNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	if len(aviList.Items) == 0 {
+		return errors.New("No AstarteVoyagerIngress resources are present in the cluster. Nothing to do here.")
+	}
+
+	// assume there is only one avi in the cluster
+	aviObject := &aviList.Items[0]
+
+	fmt.Printf("You are about to migrate the Astarte Voyager Ingress named: %s. ", aviObject.GetName())
+	shouldMigrate, err := utils.AskForConfirmation("Are you sure?")
+	if err != nil {
+		return err
+	}
+	if !shouldMigrate {
+		fmt.Println("Ok, nothing left to do here.")
+		os.Exit(0)
+	}
+
+	// if required, dump the avi custom resource
+	if fn := command.Flag("out").Value.String(); fn != "" {
+		if err := dumpResourceToYAMLFile(aviObject, fn); err != nil {
+			fmt.Println("Warning: failed to save the AstarteVoyagerIngress resource to file: ", err)
+		} else {
+			fmt.Printf("%s resource successfully written to %s\n", aviObject.GetName(), fn)
+		}
+	}
+
+	adiName, err := utils.PromptChoice("Choose the new AstarteDefaultIngress name:", "adi", false, false)
+	if err != nil {
+		return err
+	}
+
+	// We are not checking if the secret is present in the cluster: if the secret is not present, the validation webhook will return an error
```

Comment:
Is this valid for both following secrets? If so, use `secrets` instead of `the secret`